docs/source/pages/tutorials/index.rst

```diff
@@ -0,0 +1,271 @@
+
+
+This policy requires that all the dependencies
+of repository ```github.com/behnazh-w/example-maven-app``` at each commit either pass the ``mcn_provenance_level_three_1`` (have non-forgeable
+`SLSA`_ provenances) or ``mcn_infer_artifact_pipeline_1`` check. Note that if an artifact already has a non-forgeable provenance, it means it is produced
+by a hosted build platform, such as GitHub Actions CI workflows. So, the ``mcn_infer_artifact_pipeline_1`` needs to pass
+only if ``mcn_provenance_level_three_1`` fails.
+
+Let's take a closer look at this policy to understand what each line means.
+
+.. code-block:: c++
+
+  #include "prelude.dl"
+
+This line imports the predefined Datalog relations into your Datalog specification. These relations
+can be thought of as select statements specifically provided by Macaron to make it easier for you
+to write policies. In our example policy, the following relations are pre-defined:
+
+* ``Policy(policy_id: symbol, target_id: number, message: symbol)``
+* ``is_component(component_id: number, purl: symbol)``
+* ``transitive_dependency(parent: number, dependency: number)``
+* ``check_passed(component_id: number, check_name: symbol)``
+* ``apply_policy_to(policy_id: symbol, component_id: number)``
+* ``is_repo(repo_id: number, repo_complete_name: symbol, component_id: number)``
+
+And the following relation is declared in this policy:
+
+* ``violating_dependencies(parent: number)``
+
+Feel free to browse through the available
+relations `here <https://github.com/oracle/macaron/blob/main/src/macaron/policy_engine/prelude/>`_
+to see how they are constructed before moving on.
+
+.. code-block:: c++
+
+  Policy("detect-malicious-upload", component_id, "") :-
+    is_component(component_id, _),
+    !violating_dependencies(component_id).
+
+This rule populates the ``Policy`` relation if ``component_id`` exists in the database and
+``violating_dependencies`` relation for this component is empty.
+
+.. code-block:: c++
+
+  .decl violating_dependencies(parent: number)
+  violating_dependencies(parent) :-
+      transitive_dependency(parent, dependency),
+      !check_passed(dependency, "mcn_infer_artifact_pipeline_1"),
+      !check_passed(dependency, "mcn_provenance_level_three_1").
+
+This is the rule that the user needs to design to detect dependencies that violate a security requirement.
+Here we declare a relation called ``violating_dependencies`` and populate it if the dependencies in the
+``transitive_dependency`` relation do not pass any of the ``mcn_infer_artifact_pipeline_1`` and
+``mcn_provenance_level_three_1`` checks.
+
+.. code-block:: c++
+
+    apply_policy_to("detect-malicious-upload", component_id) :-
+      is_repo(_, "github.com/behnazh-w/example-maven-app", component_id).
+
+Finally, the ``apply_policy_to`` rule applies the policy ``detect-malicious-upload`` on the
+repository ``github.com/behnazh-w/example-maven-app`` on any commits available in the database.
```

Comment:
```suggestion
Finally, the ``apply_policy_to`` rule applies the policy ``detect-malicious-upload`` on any commits available in the database for the repository ``github.com/behnazh-w/example-maven-app``.
```
I think the original wordings would be confusing for user because it doesn't explicitly indicate which target the policy apply to, the commit or the repository as a whole.

Reply:
See commit [f242f41](https://github.com/oracle/macaron/pull/474/commits/f242f414c9965506a8ae55f42d22c1845624c132)